sklearn/base.py

```diff
@@ -311,16 +311,36 @@ def __setstate__(self, state):
             self.__dict__.update(state)
 
     def _get_tags(self):
+        if '_more_tags' in self.__class__.__dict__:
+            # we cannot use getattr since it would resolve to the parent's
+            # attribute if self.__class__ does not overrides it. So we use
+            # __dict__ instead.
+            overridden_tags = self._more_tags()
+        else:
+            overridden_tags = {}
+
+        def _resolve_tags(c):
+            nonlocal collected_tags
+            if c is BaseEstimator or c is object:
+                return
+
+            for direct_parent_class in c.__bases__:
+                _resolve_tags(direct_parent_class)
+                if '_more_tags' in direct_parent_class.__dict__:
+                    more_tags = direct_parent_class._more_tags(self)
+
+                    # filter-out tags that will be overridden in child class
+                    more_tags = {key: val for (key, val) in more_tags.items()
+                                 if key not in overridden_tags.keys()}
```

Comment:
So if you don't do this does the consistency check fails?

Does that generalize to multiple inheritances: A (has some tags) > B (overwrite tags) > (overwrite tags) ?

Reply:
We need this because we allow tags that are overridden on the child class to be inconsistent in the parents classes